sdk/communication/AzureCommunicationCalling/ApiSamples.md

```diff
@@ -0,0 +1,760 @@
+        self.incomingCall?.delegate = self
+    }
+
+    // Event raised when incoming call was not answered
+    public func onCallEnded(_ incomingCall: IncomingCall!, args: PropertyChangedEventArgs!) {
+        self.incomingCall = nil
+    }
+}
+```
+<!--End Start with CallClient and CallAgent-->
+
+## Call Basics
+<!--Start Call Basics-->
+### Call Properties
+Call object has various properties
+```swift
+// [String] caller identity
+self.call?.id
+
+// CallStateNone = 0,CallStateEarlyMedia = 1,CallStateIncoming = 2,CallStateConnecting = 3,CallStateRinging = 4,CallStateConnected = 5,CallStateHold = 6,CallStateDisconnecting = 7,CallStateDisconnected = 8
+self.call?.state
+
+// CallDirectionOutgoing = 1, CallDirectionIncoming = 2
+self.call?.callDirection
+
+// [Bool] isMicrophoneMuted - is local audio muted
+self.call?.isMicrophoneMuted
+
+// [Errror] callEndReason - containing code/subcode/message indicating how call ended
+self.call?.callEndReason
+
+// String[] localVideoStreams - collection of video streams send to other participants in a call
+self.call?.localVideoStreams
+
+// RemoteParticipant[] remoteParticipants - collection of remote participants participating in this call
+self.call?.remoteParticipants
+
+// Indicates if the call is currently being recorded or not
+self.call?.isRecordingActive;
+```
+
+### Call Operations
+* ### Make Outgoing Call
+
+To create and start a call you need to call one of the APIs on CallAgent
+```swift
+public class CallManager : NSObject, CallDelegate
+{
+    public func placeCall(names: [CommunicationIdentifier])
+    {
+        let callOptions = StartCallOptions()!
+        // Using the CallAgent received from the factory method
+        // and returns a Call object
+        let audioOptions = AudioOptions()
+        audioOptions!.muted = true // Start the call in mute state
+        callOptions.audioOptions = audioOptions
+        let call = self.callAgent?.startCall(participants: names,
+                                            options: callOptions)
+        call?.delegate = self.callObserver
+    }
+
+    public func placeVideoCall(names: [CommunicationIdentifier])
+    {
+        let camera = self.deviceManager?.cameras!.first
+        let localVideoStream = LocalVideoStream(camera: camera)
+        let videoOptions = VideoOptions(localVideoStream: localVideoStream)
+        
+        let callOptions = StartCallOptions()
+        callOptions?.videoOptions = videoOptions
+
+        let call = self.callAgent?.startCall(participants: names,
+                                                options: callOptions)
+        call?.delegate = self.callObserver
+    }
+
+    // Following is one event which is part of CallDelegate
+    public func OnStateChanged(_ call: Call!,
+                                args: PropertyChangedEventArgs!)
+    {
+        let state = CallObserver.callStateToString(state: call.state)
+        print("Call state changed to %@", state)
+    }
```

Comment:
Will check how we can implement this, will have to investigate. 

`OnStateChanged`  to `call` may not be possible really as it might break on other platforms and changing the logic only for iOS would be a bit complicated in ApiGen.

Reply:
BLOCKER